docker/Dockerfile

```diff
@@ -0,0 +1,39 @@
+FROM node:7
+
+# Home directory for Node-RED application source code.
+RUN mkdir -p /usr/src/node-red
+
+# User data directory, contains flows, config and nodes.
+RUN mkdir -p /data/node_modules
+
+WORKDIR /usr/src/node-red
+
+# Copying folder content to container storage
+COPY . /usr/src/node-red/
+RUN cd /usr/src/node-red/nodes/core; \
+    find -name '*.html' | \
+    grep -v -e 'template' | \
+    awk '{system("mv "$1" "$1".ign")}';
+
+# Add node-red user so we aren't running as root.
+RUN useradd --home-dir /usr/src/node-red --no-create-home node-red \
+    && chown -R node-red:node-red /data \
+    && chown -R node-red:node-red /usr/src/node-red
+
+USER node-red
+RUN npm install && npm run build
+
+# dojot customization
+COPY contrib-nodes /data/node_modules
+USER root
```

Comment:
since we're merging the two repos already, wouldn't it make sense to try and do all "privileged" work in a single block, rather than having multiple user change blocks? (remember that each instruction on the dockerfile becomes a snapshot)